src/main/java/org/folio/rest/impl/PurchaseOrderHelper.java

```diff
@@ -244,12 +237,19 @@ private int calculateTotalItemsQuantity(List<CompositePoLine> poLines) {
    * @return completable future which might be completed with {@code true} if order is valid, {@code false} if not valid or an exception if processing fails
    */
   public CompletableFuture<Boolean> validateOrder(CompositePurchaseOrder compPO) {
-    addProcessingErrors(HelperUtils.validateOrder(compPO));
 
-    // If static validation has failed, no need to call other services
-    if (!getErrors().isEmpty()) {
-      return completedFuture(false);
-    }
+    compPO.getCompositePoLines()
+      .forEach(poLine ->
+        orderLineHelper.setTenantDefaultCreateInventoryValues(poLine)
+          .thenApply(v -> {
+          addProcessingErrors(HelperUtils.validatePoLine(poLine));
```

Comment:
This `thenApply` block might be executed after `validatePoLineLimit` completes and in some case errors might be ignored...

A bit of details. 
1. The `orderLineHelper.setTenantDefaultCreateInventoryValues` will send request to `mod-configuration` and `thenApply` will be executed only after response is received.
2. `validatePoLineLimit` method below will send independent request to `mod-configuration`

In case for some reason the response for the first call will come later than for the second one, any error found in the step 1 might be missed because both completable futures are fully independent.

I simulated this by following test (which fails `Expected status code <422> but was <201>`) modifying behavior of `MockServer#handleConfigurationModuleResponse` method:

```java
...
  @Test
  public void testPostOrdersWithEmptyCreateInventory2() throws Exception {
    JsonObject order = new JsonObject(getMockData(MONOGRAPH_FOR_CREATE_INVENTORY_TEST));
    // Get Open Order
    CompositePurchaseOrder reqData = order.mapTo(CompositePurchaseOrder.class);

    // Clear CreateInventory for setting default values
    reqData.getCompositePoLines().get(0).getPhysical().setCreateInventory(null);
    reqData.getCompositePoLines().get(0).getCost().setQuantityPhysical(-1);

    verifyPostResponse(COMPOSITE_ORDERS_PATH, JsonObject.mapFrom(reqData).toString(),
      prepareHeaders(EXIST_CONFIG_X_OKAPI_TENANT_LIMIT_10), APPLICATION_JSON, 422);
  }
...
  public static class MockServer {
...
    static int delay = 1001;
    private void handleConfigurationModuleResponse(RoutingContext ctx) {
      delay -= 500;
      vertx.setTimer(delay, id -> {
        try {
          String tenant = ctx.request().getHeader(OKAPI_HEADER_TENANT) ;
          if (PO_NUMBER_ERROR_X_OKAPI_TENANT.getValue().equals(tenant)) {
            tenant = EXIST_CONFIG_X_OKAPI_TENANT_LIMIT_10.getValue();
          }
          serverResponse(ctx, 200, APPLICATION_JSON, getMockData(String.format(CONFIG_MOCK_PATH, tenant)));
        } catch (IOException e) {
          serverResponse(ctx, 500, APPLICATION_JSON, INTERNAL_SERVER_ERROR);
        }
      });
    }
```

The proposal is to move `setTenantDefaultCreateInventoryValues` block to separate method like `CompletableFuture<CompositePurchaseOrder> setTenantDefaultValues(CompositePurchaseOrder compPO)` and start any validation only on completion

Reply:
Ahh, good point, I missed this scenario

Reply:
Added a new method to handle the scenario